Feature/api v1

semensid2020 opened the pull request.

semensid2020 committed:
API v1 started
1. Controllers files created
2. Routes for namespace :api added
3. Tests for Api::V1::TasksControllerTest created

semensid2020 committed:
API v1: Various configurations and application_controller methods
1. ActiveModelSerializers set as JSON serializer
2. Ransack and Kaminari settings set at Api::V1::ApplicationController
3. Tasks serialization configured

semensid2020 committed:
API v1: INDEX SHOW CREATE methods

semensid2020 committed:
API v1: UPDATE DESTROY methods + custom responses
1. Update and destroy methods added to controller
2. JsonResponder configured
3. def api_behavior (available @JsonResponder) overridden for custom responses for PUT/PATCH:
   - updated resorce included to response

semensid2020 committed:
API v1: fixing bugs

YJorge commented on app/controllers/api/v1/application_controller.rb:
```diff
@@ -0,0 +1,32 @@
+class Api::V1::ApplicationController < Api::ApplicationController
+  respond_to :json
+  include AuthHelper
```
1. respond_to :json лучше перенести на уровень выше
2. к этому моменту пользователь уже должен быть залогинен и include AuthHelper тут не нужен

semensid2020 replied:
@YJorge 
Исправил.  Коммит API v1: fixing bugs after PR#15 review  ( **a1cd71f** )

semensid2020 committed:
API v1: fixing bugs after PR#15 review

В current_user кладём автора т.к. иначе падает ошибка:

Error:
Api::V1::TasksControllerTest#test_should_post_create:
NameError: undefined local variable or method \'current_user\' for #<Api::V1::TasksController:0x0000000000a3c0>
    app/controllers/api/v1/tasks_controller.rb:19:in \'create\'
    test/controllers/api/v1/tasks_controller_test.rb:21:in \'block in <class:TasksControllerTest>\'
